src/BaseListView.php

```diff
@@ -4,491 +4,562 @@
+    public function container(bool $value): static
     {
         $new = clone $this;
-        $new->currentPage = $currentPage;
+        $new->container = $value;
 
         return $new;
     }
 
     /**
-     * @param string $emptyText the HTML content to be displayed when {@see} does not have any data.
+     * Return a new instance with the empty text.
      *
-     * When this is set to `false` no extra HTML content will be generated.
+     * @param string $emptyText the HTML content to be displayed when {@see dataProvider} does not have any data.
      *
-     * The default value is the text "No results found." which will be translated to the current
-     * application language.
+     * The default value is the text "No results found." which will be translated to the current application language.
      *
-     * @return $this
+     * @return static
      *
-     * @see showOnEmpty
-     * @see emptyTextOptions
+     * {@see notShowOnEmpty()}
+     * {@see emptyTextAttributes()}
      */
-    public function emptyText(string $emptyText): self
+    public function emptyText(string $emptyText): static
     {
         $new = clone $this;
-
         $new->emptyText = $emptyText;
 
         return $new;
     }
 
     /**
-     * @param array $emptyTextOptions the HTML attributes for the emptyText of the list view.
+     * Returns a new instance with the HTML attributes for the empty text.
      *
-     * The "tag" element specifies the tag name of the emptyText element and defaults to "div".
+     * @param array $values Attribute values indexed by attribute names.
      *
-     * @return $this
-     *
-     * {@see Html::renderTagAttributes()} for details on how attributes are being rendered.
+     * @return static
      */
-    public function emptyTextOptions(array $emptyTextOptions): self
+    public function emptyTextAttributes(array $values): static
     {
         $new = clone $this;
-        $new->emptyTextOptions = $emptyTextOptions;
+        $new->emptyTextAttributes = $values;
 
         return $new;
     }
 
-    public function cssFramework(string $cssFramework): self
+    public function getPaginator(): PaginatorInterface
     {
-        if (!in_array($cssFramework, self::CSS_FRAMEWORKS)) {
-            $cssFramework = implode('", "', self::CSS_FRAMEWORKS);
-            throw new InvalidConfigException("Invalid CSS framework. Valid values are: \"$cssFramework\".");
+        if ($this->paginator === null) {
+            throw new InvalidArgumentException('The paginator is not set.');
         }
 
-        $new = clone $this;
-        $new->cssFramework = $cssFramework;
-
-        return $new;
+        return $this->paginator;
     }
 
-    public function getRequestArguments(): ?array
+    public function getTranslator(): TranslatorInterface
     {
-        return $this->requestArguments;
+        if ($this->translator === null) {
+            throw new InvalidArgumentException('The translator is not set.');
```

Comment:
That's runtime exception, not `InvalidArgumentException`. Also, it is a good place to apply friendly exception.

Reply:
done.